File: LiveTrackingSystem/LiveTracking.Api/Models/Shift.cs
namespace LiveTracking.Api.Models;

public class Shift
{
    public int ShiftId { get; set; }
    public string ShiftName { get; set; } = string.Empty;
    public string StartTime { get; set; } = "09:00:00"; // Format "HH:mm:ss" or "HH:mm"
    public string EndTime { get; set; } = "18:00:00";   // Format "HH:mm:ss" or "HH:mm"
    public int GracePeriodMinutes { get; set; } = 15;
    public bool IsDefault { get; set; } = false;
    public bool IsActive { get; set; } = true;
    public int? CreatedByAdminId { get; set; }
    public int? CompanyId { get; set; }
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;

    public Company? Company { get; set; }
}

File: LiveTrackingSystem/SmartWorkForce_Apps/app/src/main/java/com/zynexbd/livetracking/models/Shift.kt
package com.zynexbd.livetracking.models

import com.google.gson.annotations.SerializedName

data class Shift(
    @SerializedName("shiftId") val shiftId: Int = 0,
    @SerializedName("shiftName") val shiftName: String,
    @SerializedName("startTime") val startTime: String,
    @SerializedName("endTime") val endTime: String,
    @SerializedName("gracePeriodMinutes") val gracePeriodMinutes: Int = 15,
    @SerializedName("isDefault") val isDefault: Boolean = false,
    @SerializedName("isActive") val isActive: Boolean = true
)

data class CreateShiftRequest(
    @SerializedName("shiftName") val shiftName: String,
    @SerializedName("startTime") val startTime: String,
    @SerializedName("endTime") val endTime: String,
    @SerializedName("gracePeriodMinutes") val gracePeriodMinutes: Int = 15,
    @SerializedName("isDefault") val isDefault: Boolean = false
)

data class UpdateShiftRequest(
    @SerializedName("shiftName") val shiftName: String,
    @SerializedName("startTime") val startTime: String,
    @SerializedName("endTime") val endTime: String,
    @SerializedName("gracePeriodMinutes") val gracePeriodMinutes: Int = 15,
    @SerializedName("isDefault") val isDefault: Boolean = false,
    @SerializedName("isActive") val isActive: Boolean = true
)
